### PR TITLE
CSVFile: Discuss thread safety, misc improvements

### PR DESCRIPTION
**Pull request by davisjam**
Towards #70 and #64.

### CODE REVIEW

File: src/main/java/org/repodriller/persistence/csv/CSVFile.java [hunk 1]
```diff
@@ -16,95 +16,157 @@
...
+			try {
+				write((Object[]) header);
+			} catch (Exception e) {
+				throw new RuntimeException(e);
+			}
 		}
 	}
-	
+
+	/* Additional weird constructors. */
```
Comment (mauricioaniche) [inline src/main/java/org/repodriller/persistence/csv/CSVFile.java]: these constructors are here to maintain compatibility of very old versions of repodriller. We may mark them as deprecated in the future.

File: src/main/java/org/repodriller/persistence/csv/CSVFile.java [hunk 2]
```diff
@@ -16,95 +16,157 @@
...
-		ps.flush();
+		})
+		.toArray(String[]::new);
+
+		/* Out it goes. */
+		String csvRecord = String.join(",", escapedFields);
+		log.debug("Writing csvRecord: " + csvRecord);
+		ps.println(csvRecord);
 
+		ps.flush(); /* TODO This has performance implications. */
```
Comment (mauricioaniche) [inline src/main/java/org/repodriller/persistence/csv/CSVFile.java]: I think this dates back when repodriller crashed when an exception happens. Now that repodriller never crashes (awesome!! kk), we may leave it to the stream to decide when to flush. What do you think?

File: src/main/java/org/repodriller/persistence/csv/CSVFile.java (hunk 1 above)
Comment (davisjam) [reply src/main/java/org/repodriller/persistence/csv/CSVFile.java]: Given that you've already published old versions to Maven, is there a reason to keep deprecated methods around?

File: src/main/java/org/repodriller/persistence/csv/CSVFile.java (hunk 2 above)
Comment (davisjam) [reply src/main/java/org/repodriller/persistence/csv/CSVFile.java]: Discussion in #92

**Pull request merged**